Major fix and improvement: now work on atv 7.5.1

Belfagor2005 opened the pull request:
from feed Epgimport not work on last atv 7.5.1
I ask that the import of sources from git be taken into consideration (my git contains them and they are updated), but you can import them into yours and keep only those that are needed
From sources I have eliminated the broken links and added, thanks also to other linuxsat users, other sources.
I hope I have done something appreciated.

jbleyel commented:
Nice work but there are some issues.

You have mixed code changes and code formatting.
This is not so good because it's not very difficult to check what has been changed.

Please revert the icon because this plugin is not only for ATV.

Belfagor2005 commented:
fixed tab and space my friend (and icon)

Belfagor2005 commented:
here
https://github.com/[Belfagor2005/EPGimport-Sources](https://github.com/Belfagor2005/EPGimport-Sources)

a git for new and update source.. 
On other epgimport i have make a button for download from it.. for update.. work fine.. if you want use button blue on source for this def..

jbleyel commented:
Is this PR ready for merge? Or do you want  some more tests before we update all boxes.

Belfagor2005 commented:
You ready my last reply?  If you want fork my git source and remove Pluto and Samsung. 
I make button for git update from your repo source

jbleyel commented:
> You ready my last reply? If you want fork my git source and remove Pluto and Samsung. I make button for git update from your repo source

I'm not using this plugin.
You may should ask someone else.

jbleyel commented:
We will wait with the merge until this change is tested on other distros.

Belfagor2005 commented:
ok.. test and report first.. i make another options

Belfagor2005 commented:
As far as I'm concerned I'm done making changes, no point in continuing if the teams don't test.
I added a git import for the sources and removed some code
regards

jbleyel commented:
Running a shell script from GitHub is not really good, what exactly is this script doing?

Belfagor2005 commented:
removes sources and installs new updated ones

Belfagor2005 commented:
`https://[raw.githubusercontent.com/Belfagor2005/EPGImport-99/main/installer_source.sh`

is for all system (dreamos - and oexxx

jbleyel commented:
Why not download and extract this file:

https://github.com/Belfagor2005/EPGimport-Sources/archive/refs/heads/main.tar.gz

Using python?

AbuBaniaz commented:
Why are we renaming the custom file?

We need to avoid unnecessary differences to PLi version
https://github.com/OpenPLi/enigma2-plugin-extensions-epgimport/blob/f89622a304db7779b18f641ffd6c9427030db56b/src/EPGImport/EPGConfig.py#L166

jbleyel commented:
@Huevos, why do you have closed this pull without any comments? 
That's not polite because @Belfagor2005 has made a lot of work.

jbleyel commented:
> Why are we renaming the custom file?
> 
What custom file?

jbleyel commented:
> We need to avoid unnecessary differences to PLi version https://github.com/OpenPLi/enigma2-plugin-extensions-epgimport/blob/f89622a304db7779b18f641ffd6c9427030db56b/src/EPGImport/EPGConfig.py#L166

Since when do we take care about PLI plugins?

AbuBaniaz commented:
This plugin is forked from their copy.  The data used by it is maintained by one of the their senior people. And one of the EPG hosts is one of their developers. One of the other EPG hosts has been posting request for donations for a while now. In my opinion, we need to keep interoperability as much as possible.

Making changes that deliberately break things for no benefit whatsoever is not a good idea.

AbuBaniaz commented:
> > Why are we renaming the custom file?
> 
> What custom file?

Have a look at the changes. The current plugin across all distros uses

customFile = '/etc/epgimport/custom.channels.xml'

These commits are changing it to  
customFile = '/etc/epgimport/rytec.channels.xml'

So users will have to rename their files. Why? For what benefit? 

The plugin as it currently  works on ATV and Vix. To say it does not work is not true.  However, everything has room for improvements.

jbleyel commented:
The xml file name should not be changed.
That’s correct.
But there are some other changes and fixes what we should consider.

AbuBaniaz commented:
There is no objection from me for improvement. I object to pointless variables changes. Needless divergence from a shared codebase serves no purpose.

The way this pull request and commits could have been.....
1. Cosmetic/formatting changes
2. Description of the problem being addressed 
3. Commits to fix those problems 
4. Additional functionality as separate commits.

As stated on Skype, the grabbing sources from a web based  resource is a good addition. This is already in CrossEPG. It means we can fix things without having to wait for a package to be built. However, the sources should be where the  current sources are hosted. Not another repository.

Belfagor2005 commented:
I have no interest nor do I care to do work for you, who are also a bit rude.
You should be a bit more polite.
I think there are some differences between the team and tyrovate the hair in the egg
Meanwhile many "mine" codes have been used for improvements.
I don't even use the ATV. I don't have time for the TV and I have other things to do.
Just wasted time for you.
But you have the magic finger, you close without answering. Italians don't do that.. It must be an inherited vice.

Regards..

Huevos commented:
> @Huevos, why do you have closed this pull without any comments? That's not polite because @Belfagor2005 has made a lot of work.

I closed it so it is not merged. 

1) This commit appears to be a personal project, not for OE-A public release.
2) The sources have been changed to point at his own repo.
3) The commit is unreadable.
4) There is no proper description of what it is supposed to do.
5) It adds Python 2 code to Python 3 branch.
6) Seems to add code for other distros, no OE-A.
7) The script.
...

Belfagor2005 commented:
You are right, you are all right, but I do not know the politics of the Teams and the structure of a commit.
I have a personal git, as I said in the previous replies the sources can be where you want, I only gave the idea. I believe that you can modify/add a button to receive the updated sources.
And I am not here to argue. It seemed bad to me to receive the closure of a discussion with a response: "delete the fork if you want"
I do my best and help everyone regardless, I find many of my invented codes in many plugins. I do not live for this nor do they pay me for this, I am not a VIP in Atv or Pli, but a simple user.
What I do I do for fun and for passion. epgimport works for me .. If it does not work for others I might not care, but I receive hundreds of pm to fix it. .So, I closed the topic and deleted the discussion on linuxsa-support with this wording (it is not in the teams interest to fix epgimport".
Since there is no explanation. Who uses python 2 and is the son of another mother??

greetings and respect for you Lululla

Huevos commented:
We are all simple users here who do it for passion, no VIPs.

"Who uses python 2?" Not us, and certainly not in the python 3 branch. Your commit adds python 2 code.

You need to make "atomic" commits and describe at length what the commit does in the commit header. Don't make lots of little changes in one commit that are not related. Don't change the formatting or indent.

In our own enigma distro repos we can do what we like, but here the code is shared between all OE-A distros so we can't make wild changes that some distros might not want.

Belfagor2005 commented:
> @Belfagor2005 , is it possible to create a new pull including only the bug fix?

Check it out
I did some single pools from last night to a little while ago.
if you want we can talk oin pm on atv
You tell me what to do and I'll do it.
